substrate/frame/contracts/uapi/src/api.rs

```diff
@@ -0,0 +1,759 @@
+// Copyright (C) Parity Technologies (UK) Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// TODO:
+// - Add missing unstable methods
+
+pub mod unstable;
+use crate::{CallFlags, Result, ReturnFlags};
+use paste::paste;
+
+macro_rules! hash_fn {
+	( $name:ident, $bytes:literal ) => {
+		paste! {
+			#[doc = "Computes the " $name " " $bytes "-bit hash on the given input buffer."]
+			#[doc = "\n# Notes\n"]
+			#[doc = "- The `input` and `output` buffer may overlap."]
+			#[doc = "- The output buffer is expected to hold at least " $bytes " bits."]
+			#[doc = "- It is the callers responsibility to provide an output buffer that is large enough to hold the expected amount of bytes returned by the hash function."]
+			#[doc = "\n# Parameters\n"]
+			#[doc = "- `input`: The input data buffer."]
+			#[doc = "- `output`: The output buffer to write the hash result to."]
+			fn [<hash_ $name>](input: &[u8], output: &mut [u8; $bytes]);
+		}
+	};
+}
+
+/// Defines all the user apis implemented by both wasm and RISC-V vms.
+pub trait Api {
+	/// Returns the number of times specified contract exists on the call stack. Delegated calls are
+	/// not counted as separate calls.
+	///
+	/// # Parameters
+	///
+	/// - `account`: The contract address. Should be decodable as an `T::AccountId`. Traps
+	///   otherwise.
+	///
+	/// # Return
+	///
+	///  Returns the number of times specified contract exists on the call stack.
+	#[deprecated(note = "Unstable")]
```

Comment:
I the note should be more extensive to explain it is not really a deperecation: "Unstable function. Behaviour can change without further notice. Use only for testing".